src/providers/twitch/TwitchChannel.cpp

```diff
@@ -986,6 +993,7 @@ void TwitchChannel::createClip()
                 ->setLink(ACCOUNTS_LINK);
 
             this->addMessage(builder.release());
+            clipCreationMutex.unlock();
```

Comment:
One more thing :tm: :

I think we should consider using [QMutexLocker](https://doc.qt.io/qt-5/qmutexlocker.html) here since mutex locking/unlocking is often a source of bug when dealing with different logical branches. In fact, using a QMutexLocker is also recommended in the [QMutex docs](https://doc.qt.io/qt-5/qmutex.html#details).

Furthermore, this seems to be the Qt equivalent of the idiomatic [`std::lock_guard`](https://en.cppreference.com/w/cpp/thread/lock_guard) from the C++ standard library.

Reply:
I didn't get through the whole QMutex's documentation, thus I had no idea about this locker class existance, I will try to redo this :+1: